### PR TITLE
Refactor discovery methods

### PR DESCRIPTION
**Pull request by loay**
### Description
Refactor discovery test cases

#### Related issues
connect to https://github.com/strongloop/loopback-connector/issues/15

### CODE REVIEW

**Comment by loay** [comment]
@slnode test please

**Comment by loay** [comment]
@slnode test please

**Comment by loay** [comment]
@slnode test please

**Comment by loay** [comment]
@slnode test please

File: lib/discovery.js [hunk 1]
```diff
@@ -456,5 +389,15 @@ function mixinDiscovery(MsSQL) {
       default:
         return 'String';
     }
-  }
+  };
+
+  MsSQL.prototype.setDefaultOptions = function(options) {
```
Comment (raymondfeng) [inline lib/discovery.js]: Do we need to have the empty impls? They seem to be good candidates to have default impl in SQLConnector.

Comment (loay) [inline lib/discovery.js]: Only MySQL has a full implementation of the function so for consistency, we are throwing the error in the base connector for now, but again we will do some digging to improve this implementation in version 2.0.

**Comment by loay** [comment]
@slnode test please

**Pull request merged**